### PR TITLE
LPS-181988 Allow creating object definitions with Approved status

### PR DESCRIPTION
**Pull request by seltong**
Related issue: https://issues.liferay.com/browse/LPS-181988

### CODE REVIEW

**Comment by gabrielwas** [comment]
@seltong Can you add a simple test integration for this? Thx

**Comment by seltong** [comment]
ci:test:object

**Comment by seltong** [comment]
ci:test:relevant

**Comment by seltong** [comment]
ci:test:relevant

**Comment by seltong** [comment]
ci:test:object

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:relevant

**Comment by seltong** [comment]
ci:test:sf

File: modules/apps/object/object-admin-rest-test/src/testIntegration/java/com/liferay/object/admin/rest/resource/v1_0/test/ObjectDefinitionResourceTest.java [hunk 1]
```diff
@@ -199,6 +204,43 @@ public void testGraphQLGetObjectDefinitionsPage() throws Exception {
...
+	@Test
+	public void testPostObjectDefinition() throws Exception {
+		ObjectDefinition randomObjectDefinition = randomObjectDefinition();
+
+		ObjectDefinition postObjectDefinition =
+			testPostObjectDefinition_addObjectDefinition(
+				randomObjectDefinition);
+
+		assertEquals(randomObjectDefinition, postObjectDefinition);
+		assertValid(postObjectDefinition);
```
Comment (gabrielwas) [inline modules/apps/object/object-admin-rest-test/src/testIntegration/java/com/liferay/object/admin/rest/resource/v1_0/test/ObjectDefinitionResourceTest.java:216]: @seltong not needed (from row 210 until 216).

File: modules/apps/object/object-admin-rest-test/src/testIntegration/java/com/liferay/object/admin/rest/resource/v1_0/test/ObjectDefinitionResourceTest.java [hunk 2]
```diff
@@ -199,6 +204,43 @@ public void testGraphQLGetObjectDefinitionsPage() throws Exception {
...
+			}
+		};
+
+		randomObjectDefinition.setStatus(status);
+
+		postObjectDefinition = testPostObjectDefinition_addObjectDefinition(
+			randomObjectDefinition);
+
+		assertEquals(randomObjectDefinition, postObjectDefinition);
+		assertValid(postObjectDefinition);
```
Comment (gabrielwas) [inline modules/apps/object/object-admin-rest-test/src/testIntegration/java/com/liferay/object/admin/rest/resource/v1_0/test/ObjectDefinitionResourceTest.java:239]: assertValid not needed

File: modules/apps/object/object-admin-rest-test/src/testIntegration/java/com/liferay/object/admin/rest/resource/v1_0/test/ObjectDefinitionResourceTest.java [hunk 3]
```diff
@@ -199,6 +204,43 @@ public void testGraphQLGetObjectDefinitionsPage() throws Exception {
...
+					WorkflowConstants.getStatusLabel(0));
+			}
+		};
+
+		randomObjectDefinition.setStatus(status);
+
+		postObjectDefinition = testPostObjectDefinition_addObjectDefinition(
+			randomObjectDefinition);
+
+		assertEquals(randomObjectDefinition, postObjectDefinition);
```
Comment (gabrielwas) [inline modules/apps/object/object-admin-rest-test/src/testIntegration/java/com/liferay/object/admin/rest/resource/v1_0/test/ObjectDefinitionResourceTest.java:238]: assertEquals not needed

**Review by gabrielwas — CHANGES_REQUESTED** [review]
Check the comments. Thx.

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java [hunk 4]
```diff
@@ -369,6 +369,23 @@ public ObjectDefinition postObjectDefinition(
 				aggregationServiceBuilderObjectField.getObjectFieldSettings());
 		}
 
+		Status status = objectDefinition.getStatus();
+
+		if (FeatureFlagManagerUtil.isEnabled("LPS-167253") &&
+			(status != null) && !serviceBuilderObjectDefinition.isApproved() &&
+			(status.getCode() == WorkflowConstants.STATUS_APPROVED)) {
+
+			postObjectDefinitionPublish(
```
[the remark this answers is not in the document]
Comment (seltong) [reply modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java:378]: Sure, I sent it in the last push by mistake, sorry. I will remove it.

**Comment by seltong** [comment]
ci:test:sf

**Review by gabrielwas — APPROVED** [review]
Good job, @seltong ! LGTM :)

**Comment by seltong** [comment]
Resent here: https://github.com/brianchandotcom/liferay-portal/pull/133661

**Pull request closed without merging**